### DISCUSSION TITLE
Import custom tag from file into beets, Mediafile plugin

### DISCUSSION THREAD

**Discussion by bob-no-much-code** [beetbox/beets#5074] [Q&A] [answered]
Hello after several attempts I have managed to get the mediafile plugin to read $source from file, but I still have a couple problems.
For files where the field is undefined any query from beets returns "$source" is this the normal behaviour? How could I initialize it and set to null if undefined?
Also %ifdef($source) in the path does not seem to work. Either with the field defined or not the output is always negative.

addfield.py
```
from beets.plugins import BeetsPlugin
import mediafile

class addfield(BeetsPlugin):
    def __init__(self):
        super().__init__()
        field = mediafile.MediaField(
            mediafile.MP3DescStorageStyle(u'source'),
            mediafile.StorageStyle(u'source')
        )
        self.add_media_field('source', field)
```
path:
```
    albumtype:live: $genre/$albumartist/live/$year-$month-$day - $album %ifdef{$source} %aunique{albumartist album year, albumdisambig}%if{$albumdisambig,($albumdisambig $year) } %if{$is_1644,,%if{$is_flac,[$format $album_bitdepth-$album_samplerate],[$format $av_bitrate]}}/$disc_and_track. $title
```

**Comment by DjSlash** [comment, marked as answer]
With `%ifdef` the field should be without `$`. So: `%ifdef{source}`.

Note that you'll wind up with double spaces in your paths if the field is not defined.

**Reply by bob-no-much-code** [reply]
Thanks! I eventually found that as one of the culprits, along with me not reading carefully enough the docs and orderly changing the configs.
So what I'm using is %ifdef{source,($source) ,} and no double spaces seem to be generated.
